apps/src/netsim/NetSimRouterNode.js

```diff
@@ -606,11 +607,18 @@ NetSimRouterNode.prototype.onMessageTableChange_ = function (rows) {
   }
 };
 
+var packetEncoder = new PacketEncoder([
+  { key: 'toAddress', bits: 4 },
+  { key: 'fromAddress', bits: 4 },
+  { key: 'payload', bits: Infinity }
+]);
+
 NetSimRouterNode.prototype.routeMessage_ = function (message, myWires) {
   // Find a connection to route this message to.
-  var toAddress = message.payload.toAddress;
-  if (toAddress === undefined) {
-    //Malformed packet? Throw it away
+  try {
+    var toAddress = packetEncoder.getFieldAsInt('toAddress', message.payload);
```

Comment:
I would perhaps declare var toAddress; at the beginning of this function. How you have it should work, but I find it confusing to see vars declared in a block scope that are then used outside of that block scope.